01_literature/Engineering/Frontend/frontend-report-september-2024.md

```diff
@@ -0,0 +1,122 @@
+---
+tags:
+  - frontend
+  - market-report
+title: Frontend Report September 2024
+description: This roundup highlights the latest in React and Next.js, including Server Functions, the React 19 compiler, and self-hosting with OpenNext. Find tips on lazy loading, CSS Grid, JavaScript's evolution, plus new tools and expert insights.
+date: 2024-10-05
+authors:
+  - mashiro5951
+---
+This roundup covers the latest in React, Next.js, and web development. React Server Actions are now Server Functions, and there's a deep dive into the React 19 compiler. In Next.js news, OpenNext is enabling self-hosting, and there's fresh guidance on Progressive Web Apps. Plus, explore tips on lazy loading, CSS Grid, and JavaScript's evolution.
+
+## React
+
+### [Server Functions are Here! (They Used to be Server Actions)](https://19.react.dev/reference/rsc/server-functions "https://19.react.dev/reference/rsc/server-functions")
+
+React's Server Actions got a name change! Now they're Server Functions, and they're not just for mutations anymore.
+
+### [How to Use React Compiler – A Complete Guide](https://www.freecodecamp.org/news/react-compiler-complete-guide-react-19/)
+
+The buzz around React 19's compiler is real. This comprehensive guide walks you through everything you need to know to harness its power.
+
+### [SPA Lazy Loading: Watch Out for These Pitfalls](https://reacttraining.com/blog/spa-lazy-loading-pitfalls)
+
+Lazy loading with React Router can be tricky. Avoid common pitfalls by understanding how to structure your loader functions for optimal performance.
+
+### Quick Links
+
+- [The unspoken rules of React hooks](https://macwright.com/2024/09/19/the-extra-rules-of-hooks)
+- [React Testing Technique: Inverse Assertions](https://www.epicweb.dev/inverse-assertions)
+- [Modern Data-Fetching in React - Comparing architectures at a high level](https://reacttraining.com/blog/modern-data-fetching-in-react)
+- [Vim for React Developers - A free course focusing on real-world scenarios](https://vimforreactdevs.com/)
+
+## Next.js
```

Comment:
Seems there’s an important piece missing this month from https://www.reddit.com/r/nextjs/comments/1f92jdv/chatgptcom_switched_from_nextjs_to_remix/

Reply:
My bad, somehow I remembered it's from Aug